### ISSUE TITLE
Refactor class names used on info notices

### ISSUE THREAD

**Issue by starypop** [woocommerce/woocommerce#17555]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x] I have attempted to find the simplest possible steps to reproduce the issue

## Steps to reproduce the issue

1. Install latest WooCommerce
2. Go to My Account without making any orders
3. Visit Orders and Downloads endpoints, also try to log out.

## Expected/actual behavior

There is an inconsistency in my-account pages notice types and its CSS classes.

1) At My Account -> Downloads, when no downloads are available yet, there is a info notice with "info" icon and blue color, it has class `.woocommerce-info` (which I believe is a good behaviour).
But at My Account -> Orders, when no order has been made yet, there is a similar notice as above, but with checkmark icon (like success type), but with blue color. It's because it has both css classes: `.woocommerce-message` and `.woocommerce-info`, which I think is a bug.

2) When trying to log out from My Acoount, you'll get a notice "Are you sure you want to log out? Confirm and log out" which has type "success". But is it really a success yet? It's just a confirmation message, not fully done action. I think it should has type "notice" at this point.

3) By default a notice thrown by wc_add_notice() gets type name "success", but has a CSS class (in /templates/notices/success.php) `.woocommerce-message`. Why there is no class called `.woocommerce-succcess`? Same story with notice type "notice" having class `.woocommerce-info` instead of `.woocommerce-notice`.

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently using the steps above

**Comment by claudiulodro** [comment]
I'm sure some of these classes are there for historical reasons. We can't just change them because there are all sorts of themes and plugins that style based on the current classes, and those styles will break if we just change them.

Major releases can have breaking changes, so this might be worth standardizing then.

**Comment by starypop** [comment]
@claudiulodro I agree with you about class names and its backwards compability.

But I still believe point 1. of my previous post is a bug. 

Because in PR #12785 on line 120 someone had changed class name`.woocommerce-Message` to `.woocommerce-message` due to new naming convention. But that notice had `.woocommerce-info` class already. Now this notice has two notice type classes at the same time. This results in some kind of "hybrid" notice - its color is not corresponding with icon.

Basically we should get rid of `.woocommerce-message` class in this element, because it's overriding icon to checkmark.

What it's like now:

![zrzut ekranu 2017-11-04 o 10 23 13](https://user-images.githubusercontent.com/27000783/32404125-50182758-c14a-11e7-930e-75b9122c5a92.png)

What it should look like: 

![zrzut ekranu 2017-11-04 o 10 23 27](https://user-images.githubusercontent.com/27000783/32404124-4ffac118-c14a-11e7-88af-cda4a4d8e769.png)

**Comment by mikejolley** [comment]
That class should not be changed until we redo all of them. I think the intention is for there to be:

`woocommerce-message` as the general message class.

Then:

- `woocommerce-message-success`
- `woocommerce-message-error`
-`woocommerce-message-info`

To control styling. This has to wait for major.

**Comment by mikejolley** [comment]
See https://github.com/woocommerce/woocommerce/pull/18533
